documentation/jetty/modules/code/examples/src/main/java/org/eclipse/jetty/docs/programming/ArchitectureDocs.java

```diff
@@ -75,4 +79,60 @@ public void virtualVirtualThreads()
         HttpClient httpClient = new HttpClient(new HttpClientTransportOverHTTP(clientConnector));
         // end::virtualVirtual[]
     }
+
+    public static class EitherTask implements Invocable.Task
+    {
+        private final TryExecutor executor;
+        private final Deque<Invocable.Task> subTasks = new ConcurrentLinkedDeque<>();
+
+        public EitherTask(TryExecutor executor)
+        {
+            this.executor = executor;
+        }
+
+        @Override
+        public InvocationType getInvocationType()
+        {
+            return InvocationType.EITHER;
+        }
+
+        public void offer(Task task)
+        {
+            subTasks.add(task);
+        }
+
+        @Override
+        public void run()
+        {
+            if (Invocable.isNonBlockingInvocation())
+            {
+                for (Task subTask = subTasks.pollFirst(); subTask != null; subTask = subTasks.pollFirst())
+                {
+                    switch (Invocable.getInvocationType(subTask))
+                    {
+                        case NON_BLOCKING, EITHER -> subTask.run();
+                        case BLOCKING -> executor.execute(subTask);
+                    }
+                }
+            }
+            else
+            {
+                for (Task subTask = subTasks.pollFirst(); subTask != null; subTask = subTasks.pollFirst())
+                {
+                    switch (Invocable.getInvocationType(subTask))
+                    {
+                        case NON_BLOCKING -> subTask.run();
+                        case EITHER -> Invocable.invokeNonBlocking(subTask);
+                        case BLOCKING ->
+                        {
+                            if (executor.tryExecute(this))
+                                subTask.run();
```

Comment:
This is conceptually correct, but it has the problem that when the task finishes, the thread would loop around and iterate over the subtasks concurrently with the thread that has been asked to resume iteration via `tryExecute(this)`.

```suggestion
                            if (executor.tryExecute(this))
                            {
                                subTask.run();
                                return;
                            }
```

Reply:
In our AES we do let threads circle around and race for completing jobs, so the return doesn't reflect what we really do.   But no harm either.